File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/friend/SpouseInformationsMessage.scala





















// Generated on 05/03/2013 10:34:50
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait SpouseInformationsMessage extends MessageDeserializer {
  val messageId = 6356

  protected trait Message extends BaseMessage with Serializable {
    val messageId = SpouseInformationsMessage.messageId

    val spouse: FriendSpouseInformations.TypeImpl

    def serialize(buf: Buffer) {
      buf.writeUShort(spouse.typeId)
      spouse.serialize(buf)
    }

  }

  protected def create(spouse: FriendSpouseInformations.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val spouse = types.get[FriendSpouseInformations.TypeImpl](buf.readShort());

    create(spouse)
  }

}

object SpouseInformationsMessage extends SpouseInformationsMessage {
  case class MessageImpl(spouse: FriendSpouseInformations.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(spouse: FriendSpouseInformations.TypeImpl) = MessageImpl(spouse)

  def apply(spouse: FriendSpouseInformations.TypeImpl) = create(spouse)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/SpouseInformationsMessage.java


// Generated on 05/08/2013 19:37:51
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class SpouseInformationsMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6356;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public FriendSpouseInformations spouse;
    
    public SpouseInformationsMessage() { }
    
    public SpouseInformationsMessage(FriendSpouseInformations spouse) {
        this.spouse = spouse;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(spouse.getTypeId());
        spouse.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        spouse = ProtocolTypeManager.getInstance().build(buf.readShort());
        spouse.deserialize(buf);
    }
    
}
